services/theme_service.py

```
@@ -0,0 +1,207 @@
+
+        if mode == "submodule":
+            self._install_submodule(repo_url, name)
+        else:
+            self._install_copy(repo_url, name)
+
+        return {"name": name, "mode": mode}
+
+    def _install_submodule(self, repo_url: str, name: str) -> None:
+        """使用 git submodule add 安装主题。"""
+        try:
+            result = subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)